### PR TITLE
softether: 4.38 -> 4.41

### PR DESCRIPTION
**Pull request by 6543**
## Description of changes

Update softether package to latest stable release.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by RossComputerGuy** [comment]
Will take a look at this soon.

File: pkgs/by-name/so/softether/package.nix [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
-  build = "9760";
+  version = "4.41";
 
-  src = fetchurl {
-    url = "https://github.com/SoftEtherVPN/SoftEtherVPN_Stable/releases/download/v${version}-${build}-rtm/softether-src-v${version}-${build}-rtm.tar.gz";
-    sha256 = "0d8zahi9lkv72jh8yj66pwrsi4451vk113d3khzrzgbic6s2i0g6";
+  src = fetchFromGitHub {
+    owner = "SoftEtherVPN";
+    repo = "SoftEtherVPN_Stable";
+    rev = "v4.41-9782-beta";
```
Comment (GaetanLepage) [inline pkgs/by-name/so/softether/package.nix:19]: Shouldn't the `version` attribute equal to that ?

Comment (RossComputerGuy) [reply pkgs/by-name/so/softether/package.nix:19]: Yes, it should.

Comment (6543) [reply pkgs/by-name/so/softether/package.nix:19]: I dont mind to do so but the package version before also did not bother with that ... and upstream is kinda weard also

File: pkgs/by-name/so/softether/package.nix [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+    sha256 = "sha256-yvN5hlfAtE+gWm0s/TY/Lp53By5SDHyQIvvDutRnDNQ=";
```
Comment (GaetanLepage) [inline pkgs/by-name/so/softether/package.nix:20]:
```suggestion
    hash = "sha256-yvN5hlfAtE+gWm0s/TY/Lp53By5SDHyQIvvDutRnDNQ=";
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Could you also get rid of the `with lib;` in the `meta` attr please ?

**Comment by GaetanLepage** [comment]
> Could you also get rid of the `with lib;` in the `meta` attr please ?

I think you forgot this.
Also, in `src`, use `rev = "refs/tags/${version}";` to prevent branch/tag mismatch.

**Comment by RossComputerGuy** [comment]
Also should get rid of rec and use `finalAttrs`.

**Comment by 6543** [comment]
> I think you forgot this.

yes sorry issed that

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Comment by 6543** [comment]
> > Could you also get rid of the `with lib;` in the `meta` attr please ?

well now the build fails ... I can use the full path for e.g. maintainers ...

**Comment by GaetanLepage** [comment]
It looks like there is an issue with your change though.
They are reported to not be parseable.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 343981`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>softether</li>
  </ul>
</details>

---
### `aarch64-linux`

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Review by GaetanLepage — APPROVED** [review]
Looks good on my end.
I will let @RossComputerGuy have a final look and merge if everything is fine.

File: pkgs/by-name/so/softether/package.nix [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
   zlib,
   dataDir ? "/var/lib/softether",
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "softether";
-  version = "4.38";
-  build = "9760";
+  version = "v4.41-9782-beta";
```
Comment (RossComputerGuy) [inline pkgs/by-name/so/softether/package.nix:14]: You can remove the v and put it into the rev for the fetch.

File: pkgs/by-name/so/softether/package.nix [hunk 4]
```diff
@@ -40,11 +41,11 @@ stdenv.mkDerivation rec {
...
   '';
 
-  meta = with lib; {
+  meta = {
     description = "Open-Source Free Cross-platform Multi-protocol VPN Program";
     homepage = "https://www.softether.org/";
-    license = licenses.asl20;
-    maintainers = [ maintainers.rick68 ];
+    license = lib.licenses.asl20;
+    maintainers = [ lib.maintainers.rick68 ];
```
Comment (RossComputerGuy) [inline pkgs/by-name/so/softether/package.nix:48]: Please add `aarch64-linux` to `platforms`. Has been verified by myself on Ampere Altra Q64-22.

Comment (6543) [reply pkgs/by-name/so/softether/package.nix:48]: thanks! done

File: pkgs/by-name/so/softether/package.nix (hunk 3 above)
Comment (6543) [reply pkgs/by-name/so/softether/package.nix]: ack

**Review by RossComputerGuy — APPROVED** [review]
image0

**Comment by 6543** [comment]
:point_up: :heavy_check_mark:  that gif

**Pull request merged**